### PR TITLE
Feast Integration for simple FileSource repos

### PR DESCRIPTION
**Pull request by mathiaspet**
This PR contains the integration into Feast feature store. It contains  
- changes to configure the output of feature definitions for a file source, an entity, a feature view as well as a feature service into an existing feast repository
- unit tests for every function of the feast exporter
- a demo script to read imported data (feature_retrieval.py)
- a GH Actions workflow to export & retrieve features

### CODE REVIEW

**Comment by mathiaspet** [comment]
This PR can be reviewed and merged.

File: dev_requirements.txt [hunk 1]
```diff
@@ -5,3 +5,5 @@ pytest-cov==3.0.0
 isort==5.10.1
 flake8==4.0.1
 pre-commit==2.19.0
+pyspark==3.2.1
```
Comment (ionicsolutions) [inline dev_requirements.txt:8]:
```suggestion
```

@mathiaspet No explicit pyspark dependencies, please :-)

**Review by kajanansangar — APPROVED** [review]

**Review by kajanansangar — APPROVED** [review]

**Comment by sinuochen** [comment]
Hi @mathiaspet, Kajanan asked me to review the example notebook and I have a few comments:
* The file name should be <module_name>__<submodule_name>. In this case it can be feature_store__feast_exporter.ipynb or feature_store__feature_retrieval.ipynb
* The first cell of the notebook should be in markdown format and the table of content should reflect the content of the notbeook
* 2 packages: os and glob need to be imported
* In the last cell: write_main is not defined, I think it should be write_feast_features instead?
* Can you also run the whole notebook such that the respective outputs are reflected?
* Usually we try to cover all supported functions in the example notebook. Maybe more examples can be added.

**Comment by mathiaspet** [comment]
Hi @sinuochen, 

I addressed your comments

> * The file name should be <module_name>__<submodule_name>. In this case it can be feature_store__feast_exporter.ipynb or feature_store__feature_retrieval.ipynb
I renamed the notebook
> * The first cell of the notebook should be in markdown format and the table of content should reflect the content of the notbeook
I changed the markdown to reflect the actual content.
> * 2 packages: os and glob need to be imported
I added the imports 
> * In the last cell: write_main is not defined, I think it should be write_feast_features instead?
As mentioned in today's meeting, write_main is the actual main output function and is imported a cell before it is used.  
> * Can you also run the whole notebook such that the respective outputs are reflected?
As agreed upon in today's meeting, I skipped running the notebook. 
> * Usually we try to cover all supported functions in the example notebook. Maybe more examples can be added.
All the relevant functions are covered.

**Review by sinuochen — APPROVED** [review]

**Comment by sinuochen** [comment]
Thanks @mathiaspet, I have approved the PR.

**Pull request merged**